cypress/support/utils/check-embedded-video.js

```diff
@@ -0,0 +1,34 @@
+const selectors = {
+  fluidWidthVideoContainer: "[data-test-label='fluid-width-video-container']",
+  fluidWidthVideoWrapper: "[data-test-label='fluid-width-video-wrapper']"
+};
+
+const checkEmbeddedVideo = (headingSelector, videoSrc, i) => {
+  cy.get(headingSelector)
+    .next()
+    .next()
+    .then($el => {
+      const figure = $el[0];
+
+      expect(figure.tagName.toLowerCase()).to.equal('figure');
+      expect(figure.className).to.equal('kg-card kg-embed-card');
+      cy.wrap(figure).find(selectors.fluidWidthVideoContainer);
+      cy.wrap(figure)
+        .find(selectors.fluidWidthVideoWrapper)
+        .then($el => {
+          const fluidWidthVideoWrapper = $el[0];
+
+          expect(fluidWidthVideoWrapper.style.paddingTop).to.exist;
+        });
+      cy.wrap(figure)
+        .find('iframe')
+        .then($el => {
+          const iframe = $el[0];
+
+          expect(iframe.src).to.include(videoSrc);
+          expect(iframe.name).to.equal(`fitvid${i}`);
+        });
+    });
+};
```

Comment:
This feels like it should be part of a unit test of fitvids.

Ideally e2e tests shouldn't care about implementation details like "The sibling of my sibling should have a bunch of properties". The tests that use this probably should just check _if_ fitvids has modified a given figure not _how_ it's modified it.

What I recommend is to have one test (or test suite) that checks fitvids modifies the dom in ways you expect. Then the e2e tests can just confirm that fitvids has transformed the elements it should i.e. it's left a `data-cy="fitted"` or similar that the e2e tests can look for.

Reply:
Hey @ojeytonwilliams, thank you for pointing this out, and for your helpful suggestions. I lean on Cypress testing a bit too much for, well, everything in this repo.

Since we're no longer running fitvids in the client, I agree that it makes sense to have unit tests to check for specific structure and attributes, and e2e tests to verify that the video embeds were transformed. Also, there can be a simple e2e test to verify that video embeds weren't pushed to the bottom of the main content `div`.

It shouldn't take long to convert some of what I have here into unit tests, and simplify the e2e tests for bigger picture things.

Reply:
Sounds good!

> I lean on Cypress testing a bit too much for, well, everything in this repo.

Yeah, I generally start with Cypress tests (and sometimes stop there!). It's just that using different tools makes the separation (e2e vs unit) obvious.